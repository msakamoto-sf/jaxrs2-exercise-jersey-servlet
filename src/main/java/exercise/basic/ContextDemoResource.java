/*
 * Copyright 2014 "Masahiko Sakamoto" <sakamoto.gsyc.3s@gmail.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package exercise.basic;

import java.util.List;
import java.util.Map;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.Application;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.Cookie;
import javax.ws.rs.core.HttpHeaders;
import javax.ws.rs.core.MultivaluedMap;
import javax.ws.rs.core.PathSegment;
import javax.ws.rs.core.UriInfo;

@Path("/ex1/context-demo/{pathparam1}/{pathparam1}/{pathparam2}/{pathparam2}/{pathparam3}")
public class ContextDemoResource {

    @Context
    Application myApplication;

    @Context
    HttpHeaders httpHeaders;

    String dumpUriInfo(UriInfo info) {
        String r = "UriInfo:[\r\n";
        String abs_path = info.getAbsolutePath().toString();
        String abs_base = info.getBaseUri().toString();
        String req_uri = info.getRequestUri().toString();
        r += "AbsolutePath:'" + abs_path + "'\r\n";
        r += "BaseUri:'" + abs_base + "'\r\n";
        r += "RequestUri:'" + req_uri + "'\r\n";
        r += "QueryParameters:[\r\n";
        MultivaluedMap<String, String> queryParams = info.getQueryParameters();
        for (Map.Entry<String, List<String>> e : queryParams.entrySet()) {
            r += "  " + e.getKey() + ":[\r\n";
            List<String> values = e.getValue();
            for (String v : values) {
                r += "    " + v + "\r\n";
            }
            r += "  ]\r\n";
        }
        r += "]\r\n";
        MultivaluedMap<String, String> pathParams = info.getPathParameters();
        r += "PathParameters:[\r\n";
        for (Map.Entry<String, List<String>> e : pathParams.entrySet()) {
            r += "  " + e.getKey() + ":[\r\n";
            List<String> values = e.getValue();
            for (String v : values) {
                r += "    " + v + "\r\n";
            }
            r += "  ]\r\n";
        }
        r += "]\r\n";
        List<PathSegment> pathSegments = info.getPathSegments();
        r += "PathSegments:[\r\n";
        for (PathSegment ps : pathSegments) {
            r += "  " + ps.getPath() + "\r\n";
        }
        r += "]\r\n";
        return r + "]\r\n";
    }

    /**
     * @param info
     * @return
     */
    @GET
    @Produces("text/plain")
    public String forResourceRoot(@Context UriInfo info) {
        String classes = "Application#getClasses = [\r\n";
        for (Class<?> c : myApplication.getClasses()) {
            classes += c.getName() + "\r\n";
        }
        classes += "]\r\n";

        String headers = "HttpHeaders = [\r\n";
        MultivaluedMap<String, String> mvheaders = httpHeaders
                .getRequestHeaders();
        for (Map.Entry<String, List<String>> header : mvheaders.entrySet()) {
            headers += "  " + header.getKey() + ":[\r\n";
            List<String> values = header.getValue();
            for (String v : values) {
                headers += "    " + v + "\r\n";
            }
            headers += "  ]\r\n";
        }
        headers += "]\r\n";

        String cookies = "HttpHeaders#getCookies() = [\r\n";
        for (Map.Entry<String, Cookie> c : httpHeaders.getCookies().entrySet()) {
            cookies += "  " + c.getKey() + "=" + c.getValue().toString()
                    + "\r\n";
        }
        cookies += "]\r\n";

        return classes + headers + cookies + dumpUriInfo(info);
    }
}
