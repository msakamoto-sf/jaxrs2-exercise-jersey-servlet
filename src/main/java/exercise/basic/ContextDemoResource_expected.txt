Application#getClasses = [
...
]
HttpHeaders = [
  host:[
    127.0.0.1:8580
  ]
(...)
  cookie:[
    c1=c1v;c2=c2va;c2=c2vb;c3=c3v
  ]
  x-custom-header-without-value:[

  ]
  x-custom-header1:[
    hv1a
    hv1b
  ]
  x-custom-header2:[
    hv2
  ]
]
HttpHeaders#getCookies() = [
  c1=$Version=0;c1=c1v
  c2=$Version=0;c2=c2vb
  c3=$Version=0;c3=c3v
]
UriInfo:[
AbsolutePath:'http://127.0.0.1:8580/myapp/ex1/context-demo/pp1a/pp1b/pp2a/pp2b/pp3'
BaseUri:'http://127.0.0.1:8580/myapp/'
RequestUri:'http://127.0.0.1:8580/myapp/ex1/context-demo/pp1a/pp1b/pp2a/pp2b/pp3?q1=q1va&q1=q1vb&q1=q1vc&q2=q2va&q2=q2vb&q3=q3v'
QueryParameters:[
  q2:[
    q2va
    q2vb
  ]
  q1:[
    q1va
    q1vb
    q1vc
  ]
  q3:[
    q3v
  ]
]
PathParameters:[
  pathparam2:[
    pp2b
    pp2a
  ]
  pathparam3:[
    pp3
  ]
  pathparam1:[
    pp1b
    pp1a
  ]
]
PathSegments:[
  ex1
  context-demo
  pp1a
  pp1b
  pp2a
  pp2b
  pp3
]
]
